packages/core/src/utils.js

```diff
@@ -655,28 +656,63 @@ export function retrieveSchema(schema, rootSchema = {}, formData = {}) {
     return {};
   }
   let resolvedSchema = resolveSchema(schema, rootSchema, formData);
-  if ("allOf" in schema) {
-    try {
-      resolvedSchema = mergeAllOf({
-        ...resolvedSchema,
-        allOf: resolvedSchema.allOf,
-      });
-    } catch (e) {
-      console.warn("could not merge subschemas in allOf:\n" + e);
-      const { allOf, ...resolvedSchemaWithoutAllOf } = resolvedSchema;
-      return resolvedSchemaWithoutAllOf;
-    }
-  }
+
   const hasAdditionalProperties =
     resolvedSchema.hasOwnProperty("additionalProperties") &&
     resolvedSchema.additionalProperties !== false;
+
   if (hasAdditionalProperties) {
-    return stubExistingAdditionalProperties(
+    resolvedSchema = stubExistingAdditionalProperties(
       resolvedSchema,
       rootSchema,
       formData
     );
   }
+
+  if ("if" in resolvedSchema) {
+    // Note that if and else are key words in javascript so extract to variable names which are allowed
+    var {
+      if: expression,
+      then,
+      else: otherwise,
+      ...resolvedSchemaLessConditional
+    } = resolvedSchema;
+    var conditionalSchema = isValid(expression, formData) ? then : otherwise;
+
+    if (conditionalSchema) {
+      conditionalSchema = resolveSchema(
+        conditionalSchema,
+        rootSchema,
+        formData
+      );
+      resolvedSchema = mergeSchemas(
+        resolvedSchemaLessConditional,
+        conditionalSchema
+      );
+    }
+  }
+
+  let allOf = resolvedSchema.allOf;
+
+  if (allOf) {
+    for (var i = 0; i < allOf.length; i++) {
+      let allOfSchema = allOf[i];
+
+      // if we see an if in our all of schema then evaluate the if schema and select the then / else, not sure if we should still merge without our if then else
+      if ("if" in allOfSchema) {
+        allOfSchema = isValid(allOfSchema.if, formData)
+          ? allOfSchema.then
+          : allOfSchema.else;
+      }
+
+      if (allOfSchema) {
+        allOfSchema = resolveSchema(allOfSchema, rootSchema, formData); // resolve references etc.
+        resolvedSchema = mergeSchemas(resolvedSchema, allOfSchema);
+        delete resolvedSchema.allOf;
```

Comment:
Is it ok to mutate here? or would it be safer to do something like

`let {allOf, ...resolvedSchema = mergeSchemas(resolvedSchema, allOfSchema);`